Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -0,0 +1,351 @@
+
+            bool BeginNode(
+                void* instance, const AZ::SerializeContext::ClassData* classData, const AZ::SerializeContext::ClassElement* classElement)
+            {
+                AZStd::string path = m_stack.top().m_path;
+                if (classElement)
+                {
+                    path.append("/");
+                    path.append(classElement->m_name);
+                }
+                m_stack.push({ instance, classData ? classData->m_typeId : Uuid::CreateNull(), classData, classElement });
+                m_stack.top().m_path = AZStd::move(path);
+                CacheAttributes();
+
+                StackEntry& nodeData = m_stack.top();
+
+                if (auto handlerIt = m_handlers.find(nodeData.m_typeId); handlerIt != m_handlers.end())
+                {
+                    return handlerIt->second();
+                }
+                m_visitor->VisitObjectBegin(*this, *this);
+
+                return true;
+            }
+
+            bool EndNode()
+            {
+                StackEntry nodeData = AZStd::move(m_stack.top());
+                m_stack.pop();
+                if (auto handlerIt = m_handlers.find(nodeData.m_typeId); handlerIt != m_handlers.end())
+                {
+                    return true;
+                }
+
+                m_visitor->VisitObjectEnd();
+                return true;
+            }
+
+            const AZ::TypeId& GetType() const override
+            {
+                return m_stack.top().m_typeId;
+            }
+
+            AZStd::string_view GetTypeName() const override
+            {
+                const AZ::SerializeContext::ClassData* classData = m_serializeContext->FindClassData(m_stack.top().m_typeId);
+                return classData ? classData->m_name : "<unregistered type>";
+            }
+
+            void* Get() override
+            {
+                return m_stack.top().m_instance;
+            }
+
+            const void* Get() const override
+            {
+                return m_stack.top().m_instance;
+            }
+
+            template<typename T>
+            bool TryReadAttribute(void* instance, AZ::Attribute* attribute, Dom::Value& result) const
+            {
+                AZ::AttributeReader reader(instance, attribute);
+                T value;
+                if (reader.Read<T>(value))
+                {
+                    result = Dom::Utils::ValueFromType(value);
+                    return true;
+                }
+                return false;
+            }
+
+            template<typename First, typename Second, typename... Rest>
+            bool TryReadAttribute(void* instance, AZ::Attribute* attribute, Dom::Value& result) const
+            {
+                if (TryReadAttribute<First>(instance, attribute, result))
+                {
+                    return true;
+                }
+                return TryReadAttribute<Second, Rest...>(instance, attribute, result);
+            }
+
```

Comment:
You can use a fold expression for this as well to perform a logical or across all expanded expressions using the parameter pack.

```suggestion
            template<typename T>
            bool TryReadAttribute(void* instance, AZ::Attribute* attribute, Dom::Value& result) const
            {
                AZ::AttributeReader reader(instance, attribute);
                T value;
                if (reader.Read<T>(value))
                {
                    result = Dom::Utils::ValueFromType(value);
                    return true;
                }
                return false;
            }

            template<typename... Ts>
            bool TryReadAttributes(void* instance, AZ::Attribute* attribute, Dom::Value& result) const
            {
                return (... || TryReadAttribute<Ts...>(instance, attribute, result));
            }

```